x-pack/plugins/alerting/server/saved_objects/migrations.test.ts

```diff
@@ -571,6 +571,105 @@ describe('7.11.2', () => {
     } as SavedObjectUnsanitizedDoc<RawAlert>;
     expect(isAnyActionSupportIncidents(doc)).toBe(false);
   });
+
+  test('it does not transforms alerts when the right structure connectors is already applied', () => {
+    const migration7112 = getMigrations(encryptedSavedObjectsSetup)['7.11.2'];
+    const alert = getMockData({
+      actions: [
+        {
+          actionTypeId: '.server-log',
+          group: 'threshold met',
+          params: {
+            level: 'info',
+            message: 'log message',
+          },
+          id: '99257478-e591-4560-b264-441bdd4fe1d9',
+        },
+        {
+          actionTypeId: '.servicenow',
+          group: 'threshold met',
+          params: {
+            subAction: 'pushToService',
+            subActionParams: {
+              incident: {
+                short_description: 'SN short desc',
+                description: 'SN desc',
+                severity: '2',
+                impact: '2',
+                urgency: '2',
+              },
+              comments: [{ commentId: '1', comment: 'sn comment' }],
+            },
+          },
+          id: '1266562a-4e1f-4305-99ca-1b44c469b26e',
+        },
+      ],
+    });
+
+    expect(migration7112(alert, migrationContext)).toEqual(alert);
+  });
+
+  test('if incident attribute is an empty object, copy back the related attributes from subActionParams back to incident', () => {
+    const migration7112 = getMigrations(encryptedSavedObjectsSetup)['7.11.2'];
+    const alert = getMockData({
+      actions: [
+        {
+          actionTypeId: '.server-log',
+          group: 'threshold met',
+          params: {
+            level: 'info',
+            message: 'log message',
+          },
+          id: '99257478-e591-4560-b264-441bdd4fe1d9',
+        },
+        {
+          actionTypeId: '.servicenow',
+          group: 'threshold met',
+          params: {
+            subAction: 'pushToService',
+            subActionParams: {
+              short_description: 'SN short desc',
+              description: 'SN desc',
+              severity: '2',
+              impact: '2',
+              urgency: '2',
+              incident: {},
```

Comment:
Would we ever get in this scenario where `incident` is an empty object and the other `subActionParams` are still present? I think they'll be removed right?

Reply:
That's what @pmuellr was asking me to do. if we have user who went from 7.10.x to 7.11.0/1 and to 7.12 without noticing anything at least at this point we will bring back the connector to live as it was.

Reply:
Oh I see, so you're saying if a user goes from 7.10.x to 7.11.0, opens their previous connector and modifies it? Then the previous values will be there _and_ we'll add the newly saved fields under `incident`?

Reply:
Yes let me re-try here, If a user go from from 7.10.x to 7.11.0/1 and do not do anything and does not realize that their action are broken then if they update to 7.12.0 then everything will be back to normal